File: Python/power_of_four.py
# https://leetcode.com/problems/power-of-four/description/
from utils.examination import Examiner, Testcase


class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if 0 >= n:
            return False
        if 1 == n:
            return True
        while 1 < n:
            if 0 != n % 4:
                return False
            n = int(n / 4)
        return True

def examine()->None:
    Examiner(Solution(), 'isPowerOfFour').run((
        Testcase((16,), True),
        Testcase((1,), True),
        Testcase((5,), False),
    ))

File: TypeScript/solutions/power-of-four.ts
// https://leetcode.com/problems/power-of-four/description/
import { Examiner, Testcase } from '../utils';


function isPowerOfFour(n: number): boolean {
    if (n <= 0) return false;
    if (n === 1) return true;
    while (n > 1) {
        if (n % 4 !== 0) return false;
        n = Math.floor(n / 4);
    }
    return true;
}

export function examine(): void {
    new Examiner(isPowerOfFour).run([
        new Testcase([16], true),
        new Testcase([1], true),
        new Testcase([5], false),
    ]);
}
